DuckDuckGo/Feedback/VPNMetadataCollector.swift

```diff
@@ -220,61 +227,30 @@ final class DefaultVPNMetadataCollector: VPNMetadataCollector {
-                     lastDisconnectError: await lastDisconnectError(),
+                     lastDisconnectError: lastDisconnectError?.error,
+                     underlyingErrors: lastDisconnectError?.underlyingErrors,
                      connectedServer: connectedServer,
                      connectedServerIP: connectedServerIP)
     }
 
-    public func lastDisconnectError() async -> String {
+    public func lastDisconnectError() async -> (error: VPNMetadata.LastDisconnectError, underlyingErrors: [VPNMetadata.LastDisconnectError])? {
         if #available(iOS 16, *) {
             guard let tunnelManager = try? await NETunnelProviderManager.loadAllFromPreferences().first else {
-                return "none"
+                return nil
             }
 
-            return await withCheckedContinuation { continuation in
-                tunnelManager.connection.fetchLastDisconnectError { error in
-                    let message = {
-                        if let error = error as? NSError {
-                            if error.domain == NEVPNConnectionErrorDomain,
-                                let code = NEVPNConnectionError(rawValue: error.code) {
-                                switch code {
-                                case .overslept: return "overslept"
-                                case .noNetworkAvailable: return "noNetworkAvailable"
-                                case .unrecoverableNetworkChange: return "unrecoverableNetworkChange"
-                                case .configurationFailed: return "configurationFailed"
-                                case .serverAddressResolutionFailed: return "serverAddressResolutionFailed"
-                                case .serverNotResponding: return "serverNotResponding"
-                                case .serverDead: return "serverDead"
-                                case .authenticationFailed: return "authenticationFailed"
-                                case .clientCertificateInvalid: return "clientCertificateInvalid"
-                                case .clientCertificateNotYetValid: return "clientCertificateNotYetValid"
-                                case .clientCertificateExpired: return "clientCertificateExpired"
-                                case .pluginFailed: return "pluginFailed"
-                                case .configurationNotFound: return "configurationNotFound"
-                                case .pluginDisabled: return "pluginDisabled"
-                                case .negotiationFailed: return "negotiationFailed"
-                                case .serverDisconnected: return "serverDisconnected"
-                                case .serverCertificateInvalid: return "serverCertificateInvalid"
-                                case .serverCertificateNotYetValid: return "serverCertificateNotYetValid"
-                                case .serverCertificateExpired: return "serverCertificateExpired"
```

Comment:
I very rarely see these appear in the metadata, so I've simplified it a bit. The only consequence of this is that we'll have to look up the reason by code each time it happens, but I personally think that's fine. But, let me know if you'd rather I add some field like `codeDescription` which maps a code to its actual meaning where possible.